src/main/kotlin/org/wfanet/measurement/eventdataprovider/privacybudgetmanagement/deploy/postgres/PostgresBackingStore.kt

```diff
@@ -195,40 +217,23 @@ class PostgresBackingStoreTransactionContext(
       DO
          UPDATE SET RepetitionCount = ? + PrivacyBucketCharges.RepetitionCount;
       """.trimIndent()
-    connection.prepareStatement(insertEntrySql).use { statement: PreparedStatement ->
-      statement.setString(1, privacyBucketGroup.measurementConsumerId)
-      statement.setObject(2, privacyBucketGroup.startingDate)
-      statement.setString(3, privacyBucketGroup.ageGroup.string)
-      statement.setString(4, privacyBucketGroup.gender.string)
-      statement.setFloat(5, privacyBucketGroup.vidSampleStart)
-      statement.setFloat(6, privacyCharge.delta)
-      statement.setFloat(7, privacyCharge.epsilon)
-      statement.setInt(8, if (refundCharge) -1 else 1) // update RepetitionCount
-      // TODO(@duliomatos) Make the blocking IO run within a dispatcher using coroutines
-      statement.executeUpdate()
-    }
-  }
 
-  private fun addLedgerEntry(privacyReference: Reference) {
-    val insertEntrySql =
-      """
-        INSERT into LedgerEntries (
-          MeasurementConsumerId,
-          ReferenceId,
-          IsRefund,
-          CreateTime
-        ) VALUES (
-          ?,
-          ?,
-          ?,
-          NOW())
-      """.trimIndent()
-    connection.prepareStatement(insertEntrySql).use { statement: PreparedStatement ->
-      statement.setString(1, privacyReference.measurementConsumerId)
-      statement.setString(2, privacyReference.referenceId)
-      statement.setObject(3, privacyReference.isRefund)
-      // TODO(@duliomatos) Make the blocking IO run within a dispatcher using coroutines
-      statement.executeUpdate()
+    val statement: PreparedStatement = connection.prepareStatement(insertEntrySql)
+
+    privacyBudgetBalanceEntries.forEachIndexed { index, privacyBudgetBalanceEntry ->
+      statement.setString(1, privacyBudgetBalanceEntry.privacyBucketGroup.measurementConsumerId)
+      statement.setObject(2, privacyBudgetBalanceEntry.privacyBucketGroup.startingDate)
+      statement.setString(3, privacyBudgetBalanceEntry.privacyBucketGroup.ageGroup.string)
+      statement.setString(4, privacyBudgetBalanceEntry.privacyBucketGroup.gender.string)
+      statement.setFloat(5, privacyBudgetBalanceEntry.privacyBucketGroup.vidSampleStart)
+      statement.setFloat(6, privacyBudgetBalanceEntry.charge.delta)
+      statement.setFloat(7, privacyBudgetBalanceEntry.charge.epsilon)
+      statement.setInt(8, if (refundCharge) -1 else 1) // update RepetitionCount
+      statement.addBatch()
+      // execute every 1000 rows or less
+      if (index % 1000 == 0 || index == privacyBudgetBalanceEntries.size - 1) {
```

Comment:
Consider declaring 1000 as a manifest constant?  BTW, how was this value chosen?

Reply:
Looks like it is a suggested number, which is good enough for us : 
1) https://sqlserverguides.com/postgresql-insert-multiple-rows/#:~:text=Functions%20with%20Examples-,PostgreSQL%20INSERT%20Multiple%20Rows%20limit,INSERT%20INTO%20statement%20multiple%20times.

2) https://www.postgresql.org/docs/current/populate.html